### PR TITLE
Fix out-of-boundary grid lines in l3draw

### LINKED ISSUE #1489: `\draw_path_grid:n` draws lines outside of grid rectangle
```tex
\documentclass[varwidth, border=10pt]{standalone}
\usepackage{l3draw}

\begin{document}
\ExplSyntaxOn
\cs_new_protected:Nn \mytest_grid:nnnn
  {
    \draw_begin:
      % mark origin
      \color_fill:n { blue }
      \draw_path_circle:nn { 0pt , 0pt } { 1pt }
      \draw_path_use_clear:n { fill }
      % draw grid
      \color_stroke:n { black }
      \draw_path_grid:nnnn { #1 } { #2 } { #3 } { #4 }
      \draw_path_use_clear:n { stroke }
      % draw boundary of grid, a rectangle
      \color_stroke:n { black!20 }
      \draw_path_rectangle_corners:nn { #3 } { #4 }
      \draw_path_use_clear:n { stroke }
    \draw_end:
  }

\cs_new_protected:Nn \mytest_grids:
  {
    \mytest_grid:nnnn { 2pt } { 2pt } { 1pt , 1pt } { 11pt , 11pt } \space
    % when step > grid size, nothing should be drawn
    \mytest_grid:nnnn { 5pt } { 5pt } { 1pt , 1pt } { 5pt , 5pt }
  }

\mytest_grids: \quad Actual \par

\cs_undefine:N \__draw_path_grid_auxiii:nnnnnn
\cs_new_protected:Npn \__draw_path_grid_auxiii:nnnnnn #1#2#3#4#5#6
  {
    \__draw_path_grid_auxiv:eennnnnn
      { \fp_to_dim:n { #1 * ceil(#3/(#1)) } }
      { \fp_to_dim:n { #2 * ceil(#4/(#2)) } }
      {#1} {#2} {#3} {#4} {#5} {#6}
  }
\mytest_grids: \quad Expected
\ExplSyntaxOff
\end{document}
```

image

Pseudocode of `\draw_path_grid:n`:
```tex
\draw_path_grid:nnnn xstep ystep { x1, y1 } { x2, y2 }
  xstep = abs(xstep);
  ystep = abs(ystep);
  if (x1 > x2) then
    swap(x1, x2);
  if (y1 > y2) then
    swap(y1, y2);
  % in \@@_path_grid_auxiii:nnnnnn
  % - with trunc, x0 \in (x1 - xstep, x1]
  % - with ceil, x0 \in [x1, x1 + xstep)
  xstart = xstep * trunc(x1 / xstep);
  ystart = ystep * trunc(y1 / ystep);
  % in \@@_path_grid_auxiv:nnnnnnnn
  for (dim x = xstart; x += xstep; x <= x2)
    moveto(x, y1);
    lineto(x, y2);
  for (dim y = ystart; y += ystep; y <= y2)
    moveto(x1, y0);
    lineto(x2, y0);
```

I think the grid lines should not exceed the boundary of grid rectangle, thus the two `trunc` in `\__draw_path_grid_auxiii:nnnnnn` should both be `ceil`. I'll provide a PR.
https://github.com/latex3/latex3/blob/96bd4f36bf7b9e3238047536125b7f014efafbce/l3experimental/l3draw/l3draw-paths.dtx#L789-L795

### PR DESCRIPTION
**Pull request by muzimuzhi**
Fixes #1489.

### CODE REVIEW

File: l3experimental/l3draw/l3draw.dtx [hunk 1]
```diff
@@ -848,7 +848,7 @@
 %   This is a shortcut for \cs{draw_path_ellipse:nnn}.
 % \end{function}
 %
-% \begin{function}{\draw_path_rectangle:nn, \draw_path_rectangle_corners:nn}
+% \begin{function}[updated = 2024-03-02]{\draw_path_rectangle:nn, \draw_path_rectangle_corners:nn}
```
Comment (josephwright) [inline l3experimental/l3draw/l3draw.dtx:851]: There's no change in the documented API, so we don't normally change the dates.

File: l3experimental/l3draw/l3draw.dtx [hunk 2]
```diff
@@ -880,7 +880,7 @@
 %   \end{demo}
 % \end{function}
 %
-% \begin{function}{\draw_path_grid:nnnn}
+% \begin{function}[updated = 2024-03-02]{\draw_path_grid:nnnn}
```
Comment (josephwright) [inline l3experimental/l3draw/l3draw.dtx:883]: There's no change in the documented API, so we don't normally change the dates.

**Review by josephwright — CHANGES_REQUESTED** [review]

File: l3experimental/l3draw/l3draw.dtx (hunk 1 above)
Comment (muzimuzhi) [reply l3experimental/l3draw/l3draw.dtx]: I confused myself the 'behavior' and 'expected behavior' again. 😢 `updated` is only needed when the expected behavior changes, is it?

Comment (josephwright) [reply l3experimental/l3draw/l3draw.dtx]: @muzimuzhi  Yes, exactly

**Pull request merged**